drivers/display/ili9340/ili9340.c

```diff
@@ -0,0 +1,181 @@
+/* Copyright (c) 2017 dXplore
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ili9340_lcd.h"
+#include <drivers/display/ili9340.h>
+
+#define SYS_LOG_DOMAIN "ILI9340"
+#define SYS_LOG_LEVEL CONFIG_SYS_LOG_ILI9340_LEVEL
+#include <logging/sys_log.h>
+
+#include <gpio.h>
+#include <misc/byteorder.h>
+#include <spi.h>
+
+struct ili9340_data {
+	struct device *spi;
+	struct device *reset_gpio;
+	struct device *command_data_gpio;
+};
+
+#define ILI9340_CMD_ENTER_SLEEP 0x10
+#define ILI9340_CMD_EXIT_SLEEP 0x11
+#define ILI9340_CMD_DISPLAY_OFF 0x28
+#define ILI9340_CMD_DISPLAY_ON 0x29
+#define ILI9340_CMD_COLUMN_ADDR 0x2a
+#define ILI9340_CMD_PAGE_ADDR 0x2b
+#define ILI9340_CMD_MEM_WRITE 0x2c
+
+#define ILI9340_CMD_DATA_PIN_COMMAND 0
+#define ILI9340_CMD_DATA_PIN_DATA 1
+
+static void ili9340_exit_sleep(struct ili9340_data *data);
+static void ili9340_set_mem_area(struct ili9340_data *data, const u16_t x,
+				 const u16_t y, const u16_t w, const u16_t h);
+
+int ili9340_init(struct device *dev)
+{
+	SYS_LOG_DBG("Initializing display driver");
+
+	struct ili9340_data *data = (struct ili9340_data *)dev->driver_data;
+
+	data->spi = device_get_binding(CONFIG_ILI9340_SPI_DEV_NAME);
+	if (data->spi == NULL) {
+		SYS_LOG_ERR("Could not get SPI device for ILI9340");
+		return -EPERM;
+	}
+
+	data->reset_gpio =
+	    device_get_binding(CONFIG_ILI9340_RESET_GPIO_PORT_NAME);
+	if (data->reset_gpio == NULL) {
+		SYS_LOG_ERR("Could not get GPIO port for ILI9340 reset");
+		return -EPERM;
+	}
+	gpio_pin_configure(data->reset_gpio, CONFIG_ILI9340_RESET_PIN,
+			   GPIO_DIR_OUT);
+
+	data->command_data_gpio =
+	    device_get_binding(CONFIG_ILI9340_CMD_DATA_GPIO_PORT_NAME);
+	if (data->command_data_gpio == NULL) {
+		SYS_LOG_ERR("Could not get GPIO port for ILI9340 command/data");
+		return -EPERM;
+	}
+	gpio_pin_configure(data->command_data_gpio, CONFIG_ILI9340_CMD_DATA_PIN,
+			   GPIO_DIR_OUT);
+
+	SYS_LOG_DBG("Resetting display driver");
+	gpio_pin_write(data->reset_gpio, CONFIG_ILI9340_RESET_PIN, 1);
+	k_sleep(1);
+	gpio_pin_write(data->reset_gpio, CONFIG_ILI9340_RESET_PIN, 0);
+	k_sleep(1);
+	gpio_pin_write(data->reset_gpio, CONFIG_ILI9340_RESET_PIN, 1);
+	k_sleep(5);
+
+	SYS_LOG_DBG("Initializing LCD");
+	ili9340_lcd_init(data);
+
+	SYS_LOG_DBG("Exiting sleep mode");
+	ili9340_exit_sleep(data);
+
+	/* device_get_binding checks if driver_api is not zero before checking
+	 * device name.
+	 * So just set driver_api to 1 else the function call will fail
+	 */
+	dev->driver_api = (void *)1;
+
+	return 0;
+}
+
+void ili9340_write_pixel(const struct device *dev, const u16_t x, const u16_t y,
+			 const u8_t r, const u8_t g, const u8_t b)
+{
+	SYS_LOG_DBG("Writing pixel");
```

Comment:
Maybe you want a better message. Here it will write something like "ili9340_write_pixel: Writing pixel", sounds redundant to me. Instead you could print out the pixel references. x/y/r/g/b

Apply that comment everywhere, other debug message would gain some improvements.